lib/EnzymeTestUtils/src/finite_difference_calls.jl

```diff
@@ -0,0 +1,144 @@
+# _make_jvp_call and _wrap_function adapted from ChainRulesTestUtils
+# https://github.com/JuliaDiff/ChainRulesTestUtils.jl/blob/f76f3fc7be221e07ba9be28ef33a22238ef13661/src/finite_difference_calls.jl
+# Copyright (c) 2020 JuliaDiff
+
+#=
+    _make_jvp_call(fdm, f, rettype, y, activities)
+
+Call `FiniteDifferences.jvp` on `f` with the arguments `xs` determined by `activities`.
+
+# Arguments
+- `fdm::FiniteDifferenceMethod`: How to numerically differentiate `f`.
+- `f`: The function to differentiate.
+- `rettype`: Return activity type
+- `y`: The primal output `y=f(xs...)` or at least something of the right type.
+- `activities`: activities that would be passed to `Enzyme.autodiff`
+
+# Returns
+- `ẏ`: Derivative of output w.r.t. `t` estimated by finite differencing. If `rettype` is a
+    batch return type, then `ẏ` is a `NamedTuple` of derivatives.
+=#
+function _make_jvp_call(fdm, f, rettype, y, activities)
+    xs = map(x -> x.val, activities)
+    ẋs = map(a -> a isa Const ? nothing : a.dval, activities)
+    ignores = map(a -> a isa Const, activities)
+    f2 = _wrap_forward_function(f, xs, ignores)
+    ignores = collect(ignores)
+    if rettype <: Union{Duplicated,DuplicatedNoNeed}
+        all(ignores) && return zero_tangent(y)
+        sigargs = zip(xs[.!ignores], ẋs[.!ignores])
+        return FiniteDifferences.jvp(fdm, f2, sigargs...)
+    elseif rettype <: Union{BatchDuplicated,BatchDuplicatedNoNeed}
+        all(ignores) && return (var"1"=zero_tangent(y),)
+        sig_arg_vals = xs[.!ignores]
+        ret_dvals = map(ẋs[.!ignores]...) do sig_args_dvals...
+            FiniteDifferences.jvp(fdm, f2, zip(sig_arg_vals, sig_args_dvals)...)
+        end
+        return NamedTuple{ntuple(Symbol, length(ret_dvals))}(ret_dvals)
+    else
+        throw(ArgumentError("Unsupported return type: $rettype"))
+    end
+end
+_make_jvp_call(fdm, f, ::Type{<:Const}, y, activities) = ()
+
+#=
+    _make_j′vp_call(fdm, f, ȳ, activities)
+
+Call `FiniteDifferences.j′vp` on `f` with the arguments `xs` determined by `activities`.
+
+# Arguments
+- `fdm::FiniteDifferenceMethod`: How to numerically differentiate `f`.
+- `f`: The function to differentiate.
+- `ȳ`: The cotangent of the primal output `y=f(xs...)`.
+- `activities`: activities that would be passed to `Enzyme.autodiff`
+
+# Returns
+- `x̄s`: Derivatives of output `s` w.r.t. `xs` estimated by finite differencing.
+=#
+function _make_j′vp_call(fdm, f, ȳ, activities)
```

Comment:
Can we not have a backtick in this name?

Reply:
The backtick is present in the FiniteDifferences function it is calling, after which it is named.